tests/server/app/test_data_discovery.py

```diff
@@ -0,0 +1,72 @@
+import unittest
+
+import xcube_geodb_openeo.server.app.flask as flask_server
+import xcube_geodb_openeo.server.app.tornado as tornado_server
+import xcube_geodb_openeo.server.cli as cli
+import xcube_geodb_openeo.server.api as api
+import urllib3
+import multiprocessing
+import json
+
+from xcube_geodb_openeo.server.config import load_config
+
+
+class DataDiscoveryTest(unittest.TestCase):
+    flask = None
+    tornado = None
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        config = load_config('../../test_config.yml')
+
+        cls.flask_base_url = f'http://127.0.0.1:{cli.DEFAULT_PORT + 1}'
+
+        cls.flask = multiprocessing.Process(
+            target=flask_server.serve,
+            args=(config, '127.0.0.1', cli.DEFAULT_PORT + 1,
+                  False, False)
+        )
+        cls.flask.start()
+
+        cls.tornado_base_url = f'http://127.0.0.1:{cli.DEFAULT_PORT + 2}'
+        cls.tornado = multiprocessing.Process(
+            target=tornado_server.serve,
+            args=(config, '127.0.0.1', cli.DEFAULT_PORT + 2,
+                  False, False)
+        )
+        cls.tornado.start()
+
+        cls.http = urllib3.PoolManager()
```

Comment:
Nice setup. We should make this a base test class for the tests for the new xcube server and its extension APIs.

Reply:
Also in order to avoid code duplication with `CapabilitiesTest` above! Please fix.

Reply:
Done.